### PR TITLE
WIP: Support DPDK/SPDK in EVE

### PR DESCRIPTION
**Pull request by temerkhanov**
Support DPDK/SPDK in EVE

### CODE REVIEW

File: pkg/spdk/init.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+#Launch the VTPM server
+mkdir jail; cd jail || exit;
+/usr/bin/vtpm_server
```
Comment (rvs) [inline pkg/spdk/init.sh:5]: What's a VTPM server in this context? Does it have anything to do with Trusted Platform Module?

**Review by rvs — APPROVED** [review]
LGTM in general -- but see a tiny question.

File: pkg/spdk/init.sh (hunk 1 above)
Comment (temerkhanov) [reply pkg/spdk/init.sh:5]: It's some residue, will be removed

File: Makefile [hunk 2]
```diff
@@ -9,6 +9,7 @@ PKGBASE=github.com/lf-edge/eve
 GOMODULE=$(PKGBASE)/pkg/pillar
 GOTREE=$(CURDIR)/pkg/pillar
 PATH:=$(CURDIR)/build-tools/bin:$(PATH)
+ROOTFS_SIZE_MB:=350
```
Comment (rvs) [inline Makefile:12]: As discussed -- lets drop these changes for now. We don't need to include SPDK in the default image anyway.

File: images/rootfs.yml.in [hunk 3]
```diff
@@ -69,6 +69,9 @@ services:
      image: XENTOOLS_TAG
      cgroupsPath: /eve/services/xen-tools
      oomScoreAdj: -999
+     image: SPDK_TAG
```
Comment (rvs) [inline images/rootfs.yml.in:72]: Ditto -- lets drop this for now

File: pkg/spdk/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+WORKDIR /
+RUN git clone --depth 1 https://github.com/spdk/spdk.git --branch v${SPDK_VER}
+WORKDIR /spdk
+RUN git submodule update --init
+
+WORKDIR /
+RUN wget https://download.qemu.org/qemu-${QEMU_VER}.tar.xz 
+RUN tar xvf qemu-${QEMU_VER}.tar.xz
+
+WORKDIR /spdk/dpdk
```
Comment (rvs) [inline pkg/spdk/Dockerfile:64]: Can we please get rid of dpdk? There's really no reason we should build it -- and definitely no reason we should include it in the final package.

File: pkg/spdk/init.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+#Launch the SPDK target
+HUGEMEM=256 /usr/share/spdk/scripts/setup.sh
```
Comment (rvs) [inline pkg/spdk/init.sh:4]: Just curious -- what's this HUGEMEM knob doing?

**Review by rvs — CHANGES_REQUESTED** [review]
As we've discussed let's turn it off by default in the image, drop changes to the partition size, try to get rid of dpdk bits and also take care of a few Yetus complaints.

Other than that - this looks really good @temerkhanov -- I'd love to merge this into master soon.

**Comment by rvs** [comment]
Hey @temerkhanov -- it also breaks on aarch64

```
Step 37/65 : RUN ./configure --prefix=/usr/ --with-vhost --with-uring --with-shared &&     make && make install && make install DESTDIR=/out
 ---> Running in dd788178cf05
Using default SPDK env in /spdk/lib/env_dpdk
Using default DPDK in /spdk/dpdk/build
Configuring ISA-L (logfile: /spdk/isa-l/spdk-isal.log)...done.
Creating mk/config.mk...done.
Creating mk/cc.flags.mk...done.
Type 'make' to build.
ar: `u' modifier ignored since `D' is the default (see `U')
The Meson build system
Version: 0.54.2
Source dir: /spdk/dpdk
Build dir: /spdk/dpdk/build-tmp
Build type: native build
Program cat found: YES (/bin/cat)
Project name: DPDK
Project version: 20.08.0
Using 'CC' from environment with value: 'gcc'
Using 'CC' from environment with value: 'gcc'
C compiler for the host machine: gcc (gcc 9.3.0 "gcc (Alpine 9.3.0) 9.3.0")
C linker for the host machine: gcc ld.bfd 2.34
Host machine cpu family: aarch64
Host machine cpu: aarch64
Program pkg-config found: YES (/usr/bin/pkg-config)
Program gen-pmdinfo-cfile.sh found: YES (/spdk/dpdk/buildtools/gen-pmdinfo-cfile.sh)
Program list-dir-globs.py found: YES (/spdk/dpdk/buildtools/list-dir-globs.py)
Program check-symbols.sh found: YES (/spdk/dpdk/buildtools/check-symbols.sh)
Program options-ibverbs-static.sh found: YES (/spdk/dpdk/buildtools/options-ibverbs-static.sh)
Program binutils-avx512-check.sh found: YES (/spdk/dpdk/buildtools/binutils-avx512-check.sh)
Program python3 found: YES (/usr/bin/python3)
Program cat found: YES (/bin/cat)
Program ../buildtools/symlink-drivers-solibs.sh found: YES (/bin/sh /spdk/dpdk/config/../buildtools/symlink-drivers-solibs.sh)
Checking for size of "void *" : 8
Library m found: YES
Library numa found: NO
Found pkg-config: /usr/bin/pkg-config (1.7.2)
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libbsd found: NO (tried pkgconfig and cmake)
pcap-config found: NO
Run-time dependency pcap found: NO (tried pkgconfig and config-tool)
Compiler for C supports arguments -Wextra: YES
config/meson.build:223: WARNING: Consider using the built-in warning_level option instead of using "-Wextra".
Compiler for C supports arguments -Wcast-qual: YES
Compiler for C supports arguments -Wdeprecated: YES
Compiler for C supports arguments -Wformat-nonliteral: YES
Compiler for C supports arguments -Wformat-security: YES
Compiler for C supports arguments -Wmissing-declarations: YES
Compiler for C supports arguments -Wmissing-prototypes: YES
Compiler for C supports arguments -Wnested-externs: YES
Compiler for C supports arguments -Wold-style-definition: YES
Compiler for C supports arguments -Wpointer-arith: YES
Compiler for C supports arguments -Wsign-compare: YES
Compiler for C supports arguments -Wstrict-prototypes: YES
Compiler for C supports arguments -Wundef: YES
Compiler for C supports arguments -Wwrite-strings: YES
Compiler for C supports arguments -Wno-address-of-packed-member -Waddress-of-packed-member: YES
Compiler for C supports arguments -Wno-packed-not-aligned -Wpacked-not-aligned: YES
Compiler for C supports arguments -Wno-missing-field-initializers -Wmissing-field-initializers: YES
Program /spdk/dpdk/config/arm/armv8_machine.py found: YES (/spdk/dpdk/config/arm/armv8_machine.py)

config/arm/meson.build:156:2: ERROR: Could not execute command "/spdk/dpdk/config/arm/armv8_machine.py".
```

**Comment by rouming** [comment]
@giggsoff Petr, do you have any interest to take over and continue this work or we close? Sadly this PR does not show any performance numbers (maybe you have some?), because such a work should clearly show benefits of introducing another backend storage, otherwise the value of the whole work is just meaningless.

**Comment by rouming** [comment]
No progress on these for quite a while. @eriknordmark could you please close this with the "stalled" label?

**Comment by eriknordmark** [comment]
Closing as stalled

**Pull request closed without merging**